suite-native/graph/src/reducer.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { TimeFrameValues } from './types';
+
+interface AppGraphState {
+    selectedTimeFrame: TimeFrameValues;
+}
+
+const initialState: AppGraphState = {
+    selectedTimeFrame: 'day',
+};
+
+interface RootState {
+    appGraph: AppGraphState;
+}
+
+export const appGraphSlice = createSlice({
+    name: 'appGraph',
+    initialState,
+    reducers: {
+        setSelectedTimeFrame: (state, { payload }: PayloadAction<TimeFrameValues>) => {
+            state.selectedTimeFrame = payload;
+        },
+    },
+});
+
+export const getSelectedTimeFrame = (state: RootState) => state.appGraph.selectedTimeFrame;
```

Comment:
Perhaps we could keep selectors in a separate file?

Reply:
> Perhaps we could keep selectors in a separate file?

I think it is better to keep it together (when using the Redux Toolkit `createSlice` API). It is similar way as the ["ducks" pattern](https://github.com/erikras/ducks-modular-redux). But perhaps we should not call the file _reducer_ because, in fact, it is a slice with all its state, actions, selectors, exporting the reducer...

The original duck approach would be:

- `/features`
  - `/graph`
    - `TimeSwitchItem.tsx`: a React component that shows the UI for the selected time frame
    - `graphSlice.js`: the Redux logic for the graph feature

I am not sure if this is fully applicable for our purposes so would keep just with renaming the file:

- `/src`
    - `graphSlice.js`

And in case of more reducers in a package:

- `/src`
  - `/features`
      - `someSlice.js`
      - `someOtherDomainSlice.js`

Reply:
Yes, I agree that all the redux stuff can stay in one file. Good point with the filename as well, I will rename it to slice. 

One more benefit to what @bouchja1 just mentioned is that once that file grows too large, often it's a good indicator about splitting that slice into 2. Slices should be semantically partitioned into small chunks of feature related logic. Some slices might even be just CRUD operations (using the `entityAdapter` from redux-toolkit with some UI logic as well maybe). But once there is more than 20 properties, usually it can be divided into more slices.

Reply:
Renamed to `slice.ts` https://github.com/trezor/trezor-suite/pull/5838/commits/8d03ac13bbdc855f35c585b400951d9764b552cb

Reply:
I like what you guys said about keeping things it together. Context is important and it's even more important to not break it.

I kind of dislike filename `slice`, I agree that `reducer` is also not best, but I need to think about it.

Reply:
Honestly, I would have a `state` folder in each package with related redux logic and have `reducer`, `selectors` and `actions`. I don't like having selectors in `reducers` :( Yes, simple actions are exported from the reducer, that is true, the `actions` file would be for thunks.

Reply:
I am still going to say that thunks can be in slice as well. We actually had this format in my last job where we used redux-toolkit and it worked quite well. I would at least give it a try and we can always refactor? :) Slice is fine by me + another file for types / constants and it works pretty well imho. Of course, we can discuss it further in code improvements :)

Reply:
Actions, reducer, selectors are very closely tied and one can't exist without another one. Placing them to separate files only results to constantly switching context between these files.

Nearly always you are doing some changes in action you need to that change to reducer and vice versa. What are benefits of having multiple files that would overweight disadvantage of force people to go to two or three different files to change one thing? Or even more when you have constants file for action types in one more extra file like we do in suite.

Same goes for the the thunks, thunk is usually some code that is also in some way dependent on state of slice, selector or need to dispatch another action from thunks.

I am pretty sure that with reduced boilerplate thanks to toolkit, most of these slices files be way under 100 lines of code which could be really easily handled in one file.

Reply:
I am still for putting things together into a single slice and would say it is about getting used to a new approach with redux-toolkit. I worked with "features" approach before and it really simplifies usage. Even Mark Erickson - Redux Maintainer - [recommends](https://www.reddit.com/user/acemarke/) a single slice file... there are also links to redux docs with pointing a single file logic under this link.

Reply:
@bouchja1 I have read the recommendations and actually I do agree! The only difference is that we have a bundle per domain / page already, so it's not about putting in different files and grouping by type _(selector / reducer, etc)_. Since the Redux code is already located next to the place where it is going to be used, it is kinda like a duck. So separating into different files or not is mostly about "do we want one large file to store the code, or several smaller files and why". I don't think switching would be be a problem, while related, selectors are not always changing together with the state. You might need a new selector without changing the shape of state. Same with thunks. You might need to create a new thunk for a new feature that modifies existing state using existing action creators. Yes, it is not always the case! At times you would have to modify the reducer and then add a selector in a separate file. But imagine having a large reducer with complex thunks and and enormous switch in the middleware. I would not want to scroll 700 lines each time to reach the thunk/selector/reducer/middleware I wanna reach :D

Reply:
One domain actually doesn't mean just one reducer, sometimes you will have multiple state slices in one package. Then you will have to have multiple folders with multiple reducers/actions/selectors files and that is lot file system noise and in the end sum of lines in one of these folders don't have over 150 lines of code. 

Files are usually much smaller than 700 lines, reducers usually have like 40 lines of code and even less for selectors. Only thunks could be quite long and you can easily solve this placing reducers and selector at the of file, so you can easily navigate in file.